docs/source/user-guide/expressions.md

```diff
@@ -22,60 +22,89 @@
+| Arithmetic division       | `x / y`  | `x.div(y)`                 | `div(x, y)`                 |
+| Arithmetic remainder      | `x % y`  | `x.r#mod(y)`               | `r#mod(x, y)`               |
+| Arithmetic negation       | `-x`     | `x.neg()`                  | `neg(x, y)`                 |
+| Bitwise AND               | `x & y`  | `x.bitwise_and(y)`         | `bitwise_and(x, y)`         |
+| Bitwise OR                | `x \| y` | `x.bitwise_or(y)`          | `bitwise_or(x, y)`          |
+| Bitwise XOR               | `x ^ y`  | `x.bitwise_xor(y)`         | `bitwise_xor(x, y)`         |
+| Bitwise left shift        | `x << y` | `x.bitwise_shift_left(y)`  | `bitwise_shift_left(x, y)`  |
+| Bitwise right shift       | `x >> y` | `x.bitwise_shift_right(y)` | `bitwise_shift_right(x, y)` |
+| Logical NOT               | `!x`     | `x.not()`                  | `not(x)`                    |
+
+:::{note}
+`mod` is a reserved keyword in Rust, we use a raw identifier to name the `r#mod` function.
+:::
+
+:::{note}
+`!` is a bitwise or logical complement operator in Rust, but it only works as a logical NOT in expression API.
+:::
 
 ## Boolean Expressions
 
-| Function | Notes                                     |
-| -------- | ----------------------------------------- |
-| and      | `and(expr1, expr2)` or `expr1.and(expr2)` |
-| or       | `or(expr1, expr2)` or `expr1.or(expr2)`   |
-| not      | `not(expr)` or `expr.not()`               |
-
-## Bitwise expressions
+| Function    | Method     | Explanation |
+| ----------- | ---------- | ----------- |
+| `and(x, y)` | `x.and(y)` | Logical AND |
+| `or(x, y)`  | `x.or(y)`  | Logical OR  |
+| `not(x)`    | `x.not()`  | Logical NOT |
 
-| Function            | Notes                                                                     |
-| ------------------- | ------------------------------------------------------------------------- |
-| bitwise_and         | `bitwise_and(expr1, expr2)` or `expr1.bitwise_and(expr2)`                 |
-| bitwise_or          | `bitwise_or(expr1, expr2)` or `expr1.bitwise_or(expr2)`                   |
-| bitwise_xor         | `bitwise_xor(expr1, expr2)` or `expr1.bitwise_xor(expr2)`                 |
-| bitwise_shift_right | `bitwise_shift_right(expr1, expr2)` or `expr1.bitwise_shift_right(expr2)` |
-| bitwise_shift_left  | `bitwise_shift_left(expr1, expr2)` or `expr1.bitwise_shift_left(expr2)`   |
+:::{note}
+Since `&&` and `||` are existed as logical operators in Rust, but those are not overloadable and not works with expression API.
+:::
 
 ## Comparison Expressions
 
-| Function | Notes                 |
-| -------- | --------------------- |
-| eq       | `expr1.eq(expr2)`     |
-| gt       | `expr1.gt(expr2)`     |
-| gt_eq    | `expr1.gt_eq(expr2)`  |
-| lt       | `expr1.lt(expr2)`     |
-| lt_eq    | `expr1.lt_eq(expr2)`  |
-| not_eq   | `expr1.not_eq(expr2)` |
+| Function       | Method        | Explanation           |
```

Comment:
this might be confusing a little bit? if we compare sql and dataframe functions, we may want to name it differently? like `SQL built in function`, `DataFrame function`

Reply:
I think this might be OK because it is an `Expression API` page.
The function name between SQL built-in functions and expression functions could keep the same name for the same purpose. If not, maybe we should create a function comparison page to list differences among them.

Reply:
Id rather about giving more meaningful name, Function vs Method is confusing imho

Reply:
When I read the doc I cannot understand quickly when should I use method and when should I use function, thats my concern